onnxruntime/core/framework/sequential_executor.cc

```diff
@@ -188,6 +188,13 @@ Status SequentialExecutor::Execute(const SessionState& session_state, const std:
       profile::Color::Black);
 #endif
 
+#ifdef DEBUG_NODE_INPUTS_OUTPUTS
+    static int iteration = 0;
```

Comment:
>static int iteration [](http://example.com/codeflow?start=4&length=20)

should this counter be per session instead of per process?
AFAIK there can be multiple sessions on different threads, so there might be thread-safety concerns

Reply:
I'm not sure if that use case exists. But I suppose it's a good point. I don't see a counter for # of executions in the session state. I could add one there and increment each time ExecuteGraph is invoked.

Reply:
I added a execution counter to the session state.

Reply:
I see, not sure about adding a new data member just for this debug functionality. if it's only used in this context, how about putting the data member/member functions in an ifdef DEBUG_NODE_INPUTS_OUTPUTS?

Reply:
Yeah, so far it's just a counter for debugging. Added #ifdef around it.